dask/dataframe/groupby.py

```diff
@@ -2569,6 +2569,11 @@ def transform(self, func, *args, **kwargs):
             **kwargs,
         )
 
+        if isinstance(self, DataFrameGroupBy):
+            index_name = df3.index.name
+            df3 = df3.reset_index().set_index(index_name or "index")
```

Comment:
You can chain with ``.rename_axis(index=index_name)``

Reply:
Would be neat, but dask `DataFrame` doesn't have `rename_axis` method.

Reply:
Ah thx for pointing this out